misk-gcp/src/main/kotlin/misk/cloud/gcp/tracing/TracingLoggingEnhancer.kt

```diff
@@ -18,8 +18,8 @@ class TracingLoggingEnhancer : LoggingEnhancer {
   fun enhanceLogEntry(tracer: Tracer, builder: LogEntry.Builder) {
     val activeSpan = tracer.activeSpan()
     var traceId: String? = null
-    when (activeSpan) {
-      is DDSpan -> traceId = activeSpan.context().toTraceId()
+    when (tracer) {
+      is DDTracer -> traceId = activeSpan.context().toTraceId()
```

Comment:
hmm... if this doesn't depend on casting the Tracer, why do you even need this type switch?

Reply:
correct. cargo-culting is bad.